### PR TITLE
Fix heap overflow in oc_attachment_plugin for cuffs

### PR DESCRIPTION
**Pull request by chrissy-voir**
Split oc_attachment_plugin into two scripts moving pose and
animation handling to oc_attachment_posehandler to address
heap overflow issues.  Note: this does not address functional
deficiencies in the cuff code.

Added support for end of line comments to oc_settings.

Errata:
Cuff->Poses->Back menu item does not work correctly and needs
to be addressed.
Cuff->Poses menu does not seem to have any code to handle menu
selections

### CODE REVIEW

File: src/collar/oc_attachment_posehandler.lsl [hunk 1]
```diff
@@ -0,0 +1,439 @@
...
+
+AnimMenu(key kID, integer iAuth, string sTarget){
+    string sPrompt = "\n["+sTarget+" Poses]";
+    list lButtons = [];
+    list lUtility = [UPMENU];
+    integer i;
+    for (i=0;i<llGetListLength(g_lPoses);++i) {
+        list lPose = llParseString2List(llList2String(g_lPoses,i),["|"],[]);
+        if (llList2String(lPose,0) == sTarget) {
+            if (llListFindList(g_lActivePoses,[llList2String(g_lPoses,i)]) > -1) lButtons += [g_sChecked+llList2String(lPose,1)];
```
Comment (zontreck) [inline src/collar/oc_attachment_posehandler.lsl:291]: Here as well could be made easier to read by use of the inlined checkbox code

File: src/collar/oc_settings.lsl [hunk 2]
```diff
@@ -1,553 +1,595 @@
-// This file is part of OpenCollar.
```
Comment (zontreck) [inline src/collar/oc_settings.lsl:1]: What is the change in this file? It is showing that the entire file was changed.

File: src/collar/oc_attachment_plugin.lsl [hunk 3]
```diff
@@ -0,0 +1,658 @@
...
+        lButtons = ["Chains","Sync"];
+        if (g_bHidden) lButtons += [g_sChecked+"Hide"];
+        else lButtons += [g_sUnChecked+"Hide"];
+    } else if (sTarget == "Chains") {
+        if (g_kTexture == g_kChainTexture) lButtons = [g_sChecked+"Chain",g_sUnChecked+"Rope"];
+        else lButtons = [g_sUnChecked+"Chain",g_sChecked+"Rope"];
+        
+    } else if (sTarget == "Sync") {
+        if (g_bSyncLock) lButtons = [g_sChecked+"Sync Lock"];
+        else lButtons = [g_sUnChecked+"Sync Lock"];
```
[the remark this answers is not in the document]
Comment (chrissy-voir) [reply src/collar/oc_attachment_plugin.lsl:188]: This was added by Judge in the cuff drop I got.  I only added EOL comment handling to oc_settings.

**Comment by zontreck** [comment]
Change in oc_settings: Was the End of Line comments added by @chrissy-voir .   Allows you to put # at the end of a settings line to comment it at the end, not just a new line

**Pull request merged**